spec/app/lib/member_sync_spec.rb

```diff
@@ -0,0 +1,448 @@
+      expect(m.get_external_ids('tijuana').count).to eq(1)
+    end
+  end
+
+  context '#import_user' do
+    it 'creates a new Id member when no matching member is found' do
+      u = FactoryBot.create(:tijuana_user)
+
+      IdentityTijuana::MemberSync.import_user(u.id, 0)
+
+      expect(Member.first).to have_attributes(email: u.email)
+      expect(Member.first.get_external_ids('tijuana').count).to eq(1)
+      expect(Member.first.get_external_ids('tijuana').first)
+        .to eq(u.id.to_s)
+    end
+
+    it 'links an existing TJ user when found' do
+      u = FactoryBot.create(:tijuana_user)
+      FactoryBot.create(:member, email: u.email)
+
+      IdentityTijuana::MemberSync.import_user(u.id, 0)
+
+      expect(Member.first).to have_attributes(email: u.email)
+      expect(Member.first.get_external_ids('tijuana').count).to eq(1)
+      expect(Member.first.get_external_ids('tijuana').first)
+        .to eq(u.id.to_s)
+    end
+
+    it 'updates an existing linked TJ user' do
+      m = FactoryBot.create(:member, first_name: 'Alice')
+      u = FactoryBot.create(:tijuana_user, email: m.email, first_name: 'Bob')
+      MemberExternalId.create!(
+        system: 'tijuana', member: m, external_id: u.id.to_s
+      )
+
+      IdentityTijuana::MemberSync.import_user(u.id, 0)
+
+      expect(Member.first).to have_attributes(first_name: u.first_name)
+      expect(Member.first.get_external_ids('tijuana').count).to eq(1)
+    end
+
+    it 'skips syncing when linked, but not the primary linked user' do
+      m = FactoryBot.create(:member)
+      u1 = FactoryBot.create(
+        :tijuana_user,
+        email: 'alice@example.com',
+        first_name: 'Alice'
+      )
+      u2 = FactoryBot.create(
+        :tijuana_user,
+        email: m.email,
+        first_name: m.first_name
+      )
+      MemberExternalId.create!(
+        system: 'tijuana', member: m, external_id: u1.id.to_s
+      )
+      MemberExternalId.create!(
+        system: 'tijuana', member: m, external_id: u2.id.to_s
+      )
+
+      IdentityTijuana::MemberSync.import_user(u1.id, 0)
+
+      expect(Member.first).to have_attributes(first_name: u2.first_name)
+      expect(Member.first.get_external_ids('tijuana').count).to eq(2)
+    end
+
+    it 'skips syncing when not linked and not the primary linked user' do
+      # This is effectively checking that if a member typo's their
+      # email address in a way that the Cleanser picks up the typo and
+      # corrects it, and the corrected email points to an existing
+      # user, that the typo'ed user is not sync'ed (since it's not the
+      # primary).
+      m = FactoryBot.create(:member, email: 'alice@gmail.com', first_name: 'Alice')
+      u1 = FactoryBot.create(
+        :tijuana_user,
+        email: 'alice@gmial.com',
+        first_name: 'AL1c3'
+      )
+      u2 = FactoryBot.create(
+        :tijuana_user,
+        email: m.email,
+        first_name: m.first_name
```

Comment:
It's not really obvious that syncing was skipped, as the first name is same for `u2` and `m`.
shouldn't  `u2` name be eg. `Chalice` so that `m`'s name remains the same.

Reply:
That's the point though - u2 has the same details and indeed is linked. u1 has a typo'ed email and is not linked.

The call to `import_user` pass in u1 - so this is testing that a) u2 gets linked like it should be (indirectly though the MEI count - that could be improved, will do so) but also that u1's name was not applied to the member - it should not get synced.

I'm not sure how I can make it more clear, except possibly renaming the user vars, but you'd still have to actually clock the fact that the non-linked user is being passed in to the call being tested.

Reply:
ah, my bad! of course, we are syncing a user at a time, and here the `skips syncing when...` refers to the `u1`. makes sense... I was thinking in the background that we are iterating thru all linked users 🌧️ ...